diff -urN freetype-2.4.3.orig/builds/unix/config.sub freetype-2.4.3/builds/unix/config.sub
--- freetype-2.4.3.orig/builds/unix/config.sub	2010-10-03 19:45:06.000000000 +0100
+++ freetype-2.4.3/builds/unix/config.sub	2010-11-25 17:21:44.109591585 +0000
@@ -1325,7 +1325,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -kiwi*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
