diff -urN binutils-2.23.1.orig/bfd/config.bfd binutils-2.23.1/bfd/config.bfd
--- binutils-2.23.1.orig/bfd/config.bfd	2013-01-12 13:22:39.000000000 +0000
+++ binutils-2.23.1/bfd/config.bfd	2013-01-12 13:23:03.000000000 +0000
@@ -315,7 +315,7 @@
     ;;
   arm-*-elf | arm-*-freebsd* | arm*-*-linux-* | arm*-*-conix* | \
   arm*-*-uclinux* | arm-*-kfreebsd*-gnu | \
-  arm*-*-eabi* )
+  arm*-*-eabi* | arm*-*-kiwi*)
     targ_defvec=bfd_elf32_littlearm_vec
     targ_selvecs=bfd_elf32_bigarm_vec
     ;;
@@ -660,6 +660,11 @@
     want64=true
     targ_underscore=no
     ;;
+  x86_64-*-kiwi*)
+    targ_defvec=bfd_elf64_x86_64_vec
+    targ_selvecs="bfd_elf32_i386_vec i386pei_vec x86_64pei_vec"
+    want64=true
+    ;;
 #endif
   i[3-7]86-*-lynxos*)
     targ_defvec=bfd_elf32_i386_vec
@@ -721,6 +726,10 @@
     targ_defvec=bfd_elf32_i386_vec
     targ_selfvecs=i386chaos_vec
     ;;
+  i[3-7]86-*-kiwi*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selvecs="i386pei_vec x86_64pei_vec"
+    ;;
 
   i860-*-mach3* | i860-*-osf1* | i860-*-coff*)
     targ_defvec=i860coff_vec
diff -urN binutils-2.23.1.orig/config.sub binutils-2.23.1/config.sub
--- binutils-2.23.1.orig/config.sub	2013-01-12 13:22:40.000000000 +0000
+++ binutils-2.23.1/config.sub	2013-01-12 13:23:03.000000000 +0000
@@ -1368,7 +1368,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -kiwi*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN binutils-2.23.1.orig/configure binutils-2.23.1/configure
--- binutils-2.23.1.orig/configure	2013-01-12 13:22:40.000000000 +0000
+++ binutils-2.23.1/configure	2013-01-12 13:24:09.000000000 +0000
@@ -2869,7 +2869,7 @@
       *-*-elf* | *-*-sysv4* | *-*-unixware* | *-*-eabi* | hppa*64*-*-hpux* \
       | *-*-linux* | frv-*-uclinux* | *-*-irix5* | *-*-irix6* \
       | *-*-netbsd* | *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* \
-      | *-*-solaris2* | *-*-nto* | *-*-nacl*)
+      | *-*-solaris2* | *-*-nto* | *-*-nacl* | *-*-kiwi*)
         case "${target}" in
           *-*-linux*aout* | *-*-linux*oldld*)
             ;;
diff -urN binutils-2.23.1.orig/configure.ac binutils-2.23.1/configure.ac
--- binutils-2.23.1.orig/configure.ac	2013-01-12 13:22:40.000000000 +0000
+++ binutils-2.23.1/configure.ac	2013-01-12 13:24:04.000000000 +0000
@@ -313,7 +313,7 @@
       *-*-elf* | *-*-sysv4* | *-*-unixware* | *-*-eabi* | hppa*64*-*-hpux* \
       | *-*-linux* | frv-*-uclinux* | *-*-irix5* | *-*-irix6* \
       | *-*-netbsd* | *-*-openbsd* | *-*-freebsd* | *-*-dragonfly* \
-      | *-*-solaris2* | *-*-nto* | *-*-nacl*)
+      | *-*-solaris2* | *-*-nto* | *-*-nacl* | *-*-kiwi*)
         case "${target}" in
           *-*-linux*aout* | *-*-linux*oldld*)
             ;;
diff -urN binutils-2.23.1.orig/gas/configure.tgt binutils-2.23.1/gas/configure.tgt
--- binutils-2.23.1.orig/gas/configure.tgt	2013-01-12 13:22:40.000000000 +0000
+++ binutils-2.23.1/gas/configure.tgt	2013-01-12 13:23:03.000000000 +0000
@@ -117,6 +117,7 @@
   arm-*-eabi*)				fmt=elf em=armeabi ;;
   arm-*-symbianelf*)			fmt=elf em=symbian ;;
   arm-*-kaos*)				fmt=elf ;;
+  arm-*-kiwi*)				fmt=elf  em=armeabi ;;
   arm-*-conix*)				fmt=elf ;;
   arm-*-linux*aout*)			fmt=aout em=linux ;;
   arm-*-linux-*eabi*)			fmt=elf  em=armlinuxeabi ;;
@@ -189,6 +190,7 @@
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
+  i386-*-kiwi*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-nacl*)				fmt=elf em=nacl
     case ${cpu} in
diff -urN binutils-2.23.1.orig/ld/configure.tgt binutils-2.23.1/ld/configure.tgt
--- binutils-2.23.1.orig/ld/configure.tgt	2013-01-12 13:22:42.000000000 +0000
+++ binutils-2.23.1/ld/configure.tgt	2013-01-12 13:23:03.000000000 +0000
@@ -78,6 +78,7 @@
 	  		targ_emul=armelf ;;
 arm*-*-symbianelf*)     targ_emul=armsymbian;;
 arm-*-kaos*)		targ_emul=armelf ;;
+arm*-*-kiwi*)		targ_emul=armelf ;;
 arm9e-*-elf)		targ_emul=armelf ;;
 arm*b-*-linux-*eabi*)	targ_emul=armelfb_linux_eabi
 			targ_extra_emuls=armelf_linux_eabi
@@ -261,6 +262,10 @@
 			targ_extra_emuls="elf_i386 elf_l1om elf_k1om"
 			;;
 i[3-7]86-*-kaos*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-kiwi*)	targ_emul=elf_i386 ;;
+x86_64-*-kiwi*)		targ_emul=elf_x86_64
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-freebsdaout* | i[3-7]86-*-freebsd[12].* | i[3-7]86-*-freebsd[12])
 			targ_emul=i386bsd ;;
 i[3-7]86-*-dragonfly*)	targ_emul=elf_i386
