diff -urN compiler-rt-3.3.src.orig/lib/arm/Makefile.mk compiler-rt-3.3.src/lib/arm/Makefile.mk
--- compiler-rt-3.3.src.orig/lib/arm/Makefile.mk	2013-07-16 17:01:30.000000000 +0100
+++ compiler-rt-3.3.src/lib/arm/Makefile.mk	2013-07-16 17:02:28.000000000 +0100
@@ -9,7 +9,7 @@
 
 ModuleName := builtins
 SubDirs := 
-OnlyArchs := armv5 armv6 armv7 armv7f armv7k armv7s
+OnlyArchs := arm armv5 armv6 armv7 armv7f armv7k armv7s
 
 AsmSources := $(foreach file,$(wildcard $(Dir)/*.S),$(notdir $(file)))
 Sources := $(foreach file,$(wildcard $(Dir)/*.c),$(notdir $(file)))
diff -urN compiler-rt-3.3.src.orig/lib/int_endianness.h compiler-rt-3.3.src/lib/int_endianness.h
--- compiler-rt-3.3.src.orig/lib/int_endianness.h	2013-07-16 17:01:30.000000000 +0100
+++ compiler-rt-3.3.src/lib/int_endianness.h	2013-07-16 17:02:28.000000000 +0100
@@ -93,6 +93,20 @@
 
 #endif /* GNU/Linux */
 
+/* .. */
+
+#if defined(__Kiwi__)
+
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define _YUGA_LITTLE_ENDIAN 0
+#define _YUGA_BIG_ENDIAN    1
+#elif __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define _YUGA_LITTLE_ENDIAN 1
+#define _YUGA_BIG_ENDIAN    0
+#endif
+
+#endif /* Kiwi */
+
 #if defined(_WIN32)
 
 #define _YUGA_LITTLE_ENDIAN 1
diff -urN compiler-rt-3.3.src.orig/make/platform/clang_kiwi.mk compiler-rt-3.3.src/make/platform/clang_kiwi.mk
--- compiler-rt-3.3.src.orig/make/platform/clang_kiwi.mk	1970-01-01 01:00:00.000000000 +0100
+++ compiler-rt-3.3.src/make/platform/clang_kiwi.mk	2013-07-16 17:02:28.000000000 +0100
@@ -0,0 +1,63 @@
+Description := Static runtime libraries for clang/Linux.
+
+###
+
+CC := clang
+Arch := unknown
+Configs :=
+
+# We don't currently have any general purpose way to target architectures other
+# than the compiler defaults (because there is no generalized way to invoke
+# cross compilers). For now, we just find the target archicture of the compiler
+# and only define configurations we know that compiler can generate.
+CompilerTargetTriple := $(shell \
+	$(CC) -v 2>&1 | grep 'Target:' | cut -d' ' -f2)
+ifneq ($(DEBUGMAKE),)
+ifeq ($(CompilerTargetTriple),)
+$(error "unable to infer compiler target triple for $(CC)")
+endif
+endif
+
+# Only define configs if we detected a Kiwi target.
+ifneq ($(findstring -kiwi,$(CompilerTargetTriple)),)
+
+Configs := $(firstword $(subst -, ,$(CompilerTargetTriple)))
+
+# Build 32-bit libraries on x86_64.
+ifeq ($(Configs),x86_64)
+  Configs += i386
+endif
+
+endif
+
+
+###
+
+
+CFLAGS := -Wall -Werror -O3 -fomit-frame-pointer --sysroot=$(ProjSrcRoot)/SDKs/linux
+
+# Configuration for i386.
+Arch.i386 := i386
+CFLAGS.i386 := $(CFLAGS) -m32
+FUNCTIONS.i386 := $(CommonFunctions) $(ArchFunctions.i386)
+
+# Configuration for x86_64.
+Arch.x86_64 := x86_64
+CFLAGS.x86_64 := $(CFLAGS) -m64
+FUNCTIONS.x86_64 := $(CommonFunctions) $(ArchFunctions.x86_64)
+
+# Configuration for ARM.
+Arch.arm := arm
+CFLAGS.arm = $(CFLAGS) -march=armv6 -mfpu=vfp -mfloat-abi=softfp
+FUNCTIONS.arm = $(call set_difference, $(CommonFunctions), clear_cache) \
+	aeabi_ldivmod \
+	aeabi_uldivmod \
+	aeabi_memset \
+	aeabi_memmove \
+	aeabi_memcpy
+
+# Always use optimized variants.
+OPTIMIZED := 1
+
+# We don't need to use visibility hidden on Linux.
+VISIBILITY_HIDDEN := 0
