diff -urN gcc-4.6.2.orig/config.sub gcc-4.6.2/config.sub
--- gcc-4.6.2.orig/config.sub	2011-11-19 10:30:39.000000000 +0000
+++ gcc-4.6.2/config.sub	2011-11-19 10:33:37.000000000 +0000
@@ -1317,7 +1317,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -kiwi*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN gcc-4.6.2.orig/gcc/config/arm/kiwi.h gcc-4.6.2/gcc/config/arm/kiwi.h
--- gcc-4.6.2.orig/gcc/config/arm/kiwi.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.6.2/gcc/config/arm/kiwi.h	2011-11-19 10:33:38.000000000 +0000
@@ -0,0 +1,46 @@
+/* Definitions for Kiwi ARM targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+#undef  TARGET_VERSION
+#define TARGET_VERSION fprintf (stderr, " (ARM Kiwi/ELF)");
+
+/* We want both the BPABI builtins and the Kiwi builtins. */
+#undef  TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS() 		\
+  do 						\
+    {						\
+      TARGET_BPABI_CPP_BUILTINS();		\
+      KIWI_TARGET_OS_CPP_BUILTINS();		\
+    }						\
+  while (false)
+
+#define KIWI_LINK_SPEC " \
+  %{shared:-shared} \
+  %{symbolic:-Bsymbolic} \
+  %{mbig-endian:-EB} %{mlittle-endian:-EL} \
+  -X \
+  %{!shared: \
+    %{!static: \
+      %{rdynamic:-export-dynamic} \
+      %{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}} \
+    %{static:-Bstatic}}"
+
+/* Include BPABI link spec as well as our own. */
+#undef  LINK_SPEC
+#define LINK_SPEC KIWI_LINK_SPEC BE8_LINK_SPEC
diff -urN gcc-4.6.2.orig/gcc/config/arm/t-kiwi gcc-4.6.2/gcc/config/arm/t-kiwi
--- gcc-4.6.2.orig/gcc/config/arm/t-kiwi	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.6.2/gcc/config/arm/t-kiwi	2011-11-19 10:33:38.000000000 +0000
@@ -0,0 +1,2 @@
+# These functions are included in shared libraries.
+TARGET_LIBGCC2_CFLAGS = -fPIC
diff -urN gcc-4.6.2.orig/gcc/config/i386/kiwi32.h gcc-4.6.2/gcc/config/i386/kiwi32.h
--- gcc-4.6.2.orig/gcc/config/i386/kiwi32.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.6.2/gcc/config/i386/kiwi32.h	2011-11-19 10:33:38.000000000 +0000
@@ -0,0 +1,31 @@
+/* Definitions for Kiwi i386 targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+#undef  TARGET_VERSION
+#define TARGET_VERSION fprintf (stderr, " (i386 Kiwi/ELF)");
+
+#undef  LINK_SPEC
+#define LINK_SPEC "%{m64:-m elf_x86_64} %{!m64:-m elf_i386} \
+  %{shared:-shared} \
+  %{!shared: \
+    %{!static: \
+      %{rdynamic:-export-dynamic} \
+      %{m64:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}} \
+      %{!m64:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}}} \
+    %{static:-static}}"
diff -urN gcc-4.6.2.orig/gcc/config/i386/kiwi64.h gcc-4.6.2/gcc/config/i386/kiwi64.h
--- gcc-4.6.2.orig/gcc/config/i386/kiwi64.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.6.2/gcc/config/i386/kiwi64.h	2011-11-19 10:33:38.000000000 +0000
@@ -0,0 +1,31 @@
+/* Definitions for Kiwi x86-64 targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+#undef  TARGET_VERSION
+#define TARGET_VERSION fprintf (stderr, " (x86-64 Kiwi/ELF)");
+
+#undef  LINK_SPEC
+#define LINK_SPEC "%{!m32:-m elf_x86_64} %{m32:-m elf_i386} \
+  %{shared:-shared} \
+  %{!shared: \
+    %{!static: \
+      %{rdynamic:-export-dynamic} \
+      %{!m32:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}} \
+      %{m32:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}}} \
+    %{static:-static}}"
diff -urN gcc-4.6.2.orig/gcc/config/i386/t-kiwi64 gcc-4.6.2/gcc/config/i386/t-kiwi64
--- gcc-4.6.2.orig/gcc/config/i386/t-kiwi64	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.6.2/gcc/config/i386/t-kiwi64	2011-11-19 10:33:38.000000000 +0000
@@ -0,0 +1,17 @@
+MULTILIB_OPTIONS = m64/m32
+MULTILIB_DIRNAMES = 64 32
+MULTILIB_OSDIRNAMES = ../lib ../lib32
+
+LIBGCC = stmp-multilib
+INSTALL_LIBGCC = install-multilib
+   
+EXTRA_MULTILIB_PARTS=crtbegin.o crtend.o
+
+# The pushl in CTOR initialization interferes with frame pointer elimination.
+# crtend*.o cannot be compiled without -fno-asynchronous-unwind-tables,
+# because then __FRAME_END__ might not be the last thing in .eh_frame
+# section.
+CRTSTUFF_T_CFLAGS = -fno-omit-frame-pointer -fno-asynchronous-unwind-tables
+
+# Compile libgcc2.a with pic.
+TARGET_LIBGCC2_CFLAGS = -fPIC
diff -urN gcc-4.6.2.orig/gcc/config/kiwi.h gcc-4.6.2/gcc/config/kiwi.h
--- gcc-4.6.2.orig/gcc/config/kiwi.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.6.2/gcc/config/kiwi.h	2011-11-19 10:33:38.000000000 +0000
@@ -0,0 +1,44 @@
+/* Definitions for Kiwi targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+/* Specify predefined symbols in preprocessor.  */
+#undef  KIWI_TARGET_OS_CPP_BUILTINS
+#define KIWI_TARGET_OS_CPP_BUILTINS()		\
+	do {					\
+		builtin_define ("__Kiwi__");	\
+	} while (0)
+
+#undef  TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()		\
+	KIWI_TARGET_OS_CPP_BUILTINS()
+
+#undef  STARTFILE_SPEC
+#define STARTFILE_SPEC \
+  "%{!shared: %{pg|p|profile:gcrt1.o%s;pie:Scrt1.o%s;:crt1.o%s}} \
+   crti.o%s %{static:crtbeginT.o%s;shared|pie:crtbeginS.o%s;:crtbegin.o%s}"
+
+#undef  ENDFILE_SPEC
+#define ENDFILE_SPEC \
+  "%{shared|pie:crtendS.o%s;:crtend.o%s} crtn.o%s"
+
+#undef  LIB_SPEC
+#define LIB_SPEC "-lc"
+
+/* Make no assumptions about header files. */
+#define NO_IMPLICIT_EXTERN_C
diff -urN gcc-4.6.2.orig/gcc/config.gcc gcc-4.6.2/gcc/config.gcc
--- gcc-4.6.2.orig/gcc/config.gcc	2011-11-19 10:30:41.000000000 +0000
+++ gcc-4.6.2/gcc/config.gcc	2011-11-19 10:33:38.000000000 +0000
@@ -700,6 +700,13 @@
   default_use_cxa_atexit=yes
   use_gcc_stdint=wrap
   ;;
+*-*-kiwi*)
+  extra_parts="crtbegin.o crtend.o crtbeginS.o crtbeginT.o crtendS.o"
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  tmake_file="t-svr4"
+  ;;
 esac
 
 case ${target} in
@@ -895,6 +902,11 @@
 	tm_file="${tm_file} arm/aout.h arm/arm.h"
 	tmake_file="${tmake_file} arm/t-arm-softfp soft-fp/t-softfp"
 	;;
+arm*-*-kiwi*)
+	tm_file="dbxelf.h elfos.h kiwi.h arm/elf.h arm/bpabi.h arm/kiwi.h arm/aout.h arm/arm.h"
+	tmake_file="${tmake_file} arm/t-arm arm/t-arm-elf arm/t-bpabi arm/t-arm-softfp soft-fp/t-softfp"
+	need_64bit_hwint=yes
+	;;
 arm*-*-rtems*)
 	tm_file="dbxelf.h elfos.h arm/unknown-elf.h arm/elf.h arm/aout.h arm/arm.h arm/rtems-elf.h rtems.h newlib-stdint.h"
 	tmake_file="arm/t-arm arm/t-arm-elf t-rtems arm/t-rtems"
@@ -1593,6 +1605,14 @@
 		tm_file="${tm_file} dbxcoff.h"
 	fi
 	;;
+i[34567]86-*-kiwi*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h kiwi.h i386/kiwi32.h"
+	tmake_file="${tmake_file} i386/t-crtstuff"
+	;;
+x86_64-*-kiwi*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h kiwi.h i386/x86-64.h i386/kiwi64.h"
+	tmake_file="${tmake_file} i386/t-kiwi64"
+	;;
 ia64*-*-elf*)
 	tm_file="${tm_file} dbxelf.h elfos.h newlib-stdint.h ia64/sysv4.h ia64/elf.h"
 	tmake_file="ia64/t-ia64"
diff -urN gcc-4.6.2.orig/libgcc/config.host gcc-4.6.2/libgcc/config.host
--- gcc-4.6.2.orig/libgcc/config.host	2011-11-19 10:30:43.000000000 +0000
+++ gcc-4.6.2/libgcc/config.host	2011-11-19 10:33:38.000000000 +0000
@@ -218,6 +218,8 @@
 	;;
 arm*-*-eabi* | arm*-*-symbianelf* )
 	;;
+arm*-*-kiwi*)
+	;;
 arm*-*-rtems*)
 	;;
 arm*-*-elf)
@@ -281,6 +283,10 @@
 	;;
 x86_64-*-elf*)
 	;;
+i[34567]86-*-kiwi*)
+	;;
+x86_64-*-kiwi*)
+	;;
 i[34567]86-*-freebsd*)
 	tmake_file="${tmake_file} i386/t-freebsd"
 	;;
diff -urN gcc-4.6.2.orig/libstdc++-v3/configure.host gcc-4.6.2/libstdc++-v3/configure.host
--- gcc-4.6.2.orig/libstdc++-v3/configure.host	2011-11-19 10:30:44.000000000 +0000
+++ gcc-4.6.2/libstdc++-v3/configure.host	2011-11-19 10:33:38.000000000 +0000
@@ -253,6 +253,10 @@
     atomicity_dir=os/irix
     atomic_word_dir=os/irix
     ;;
+  kiwi*)
+    error_constants_dir="os/generic"
+    os_include_dir="os/generic"
+    ;;
   mingw32*)
     os_include_dir="os/mingw32"
     error_constants_dir="os/mingw32"
diff -urN gcc-4.6.2.orig/libstdc++-v3/crossconfig.m4 gcc-4.6.2/libstdc++-v3/crossconfig.m4
--- gcc-4.6.2.orig/libstdc++-v3/crossconfig.m4	2011-11-19 10:30:44.000000000 +0000
+++ gcc-4.6.2/libstdc++-v3/crossconfig.m4	2011-11-19 10:33:38.000000000 +0000
@@ -141,6 +141,10 @@
 	;;
     esac
     ;;
+  *-kiwi*)
+    GLIBCXX_CHECK_COMPILER_FEATURES
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    ;;
   *-linux* | *-uclinux* | *-gnu* | *-kfreebsd*-gnu | *-knetbsd*-gnu)
     GLIBCXX_CHECK_COMPILER_FEATURES
     GLIBCXX_CHECK_LINKER_FEATURES
diff -urN gcc-4.6.2.orig/libtool.m4 gcc-4.6.2/libtool.m4
--- gcc-4.6.2.orig/libtool.m4	2011-11-19 10:30:49.000000000 +0000
+++ gcc-4.6.2/libtool.m4	2011-11-19 10:33:38.000000000 +0000
@@ -5362,7 +5362,9 @@
   _LT_LINKER_SHLIBS($1)
   _LT_SYS_DYNAMIC_LINKER($1)
   _LT_LINKER_HARDCODE_LIBPATH($1)
-  LT_SYS_DLOPEN_SELF
+  if test "$cross_compiling" = no; then
+    LT_SYS_DLOPEN_SELF
+  fi
   _LT_CMD_STRIPLIB
 
   # Report which library types will actually be built
diff -urN gcc-4.6.2.orig/ltmain.sh gcc-4.6.2/ltmain.sh
--- gcc-4.6.2.orig/ltmain.sh	2011-11-19 10:30:49.000000000 +0000
+++ gcc-4.6.2/ltmain.sh	2011-11-19 10:33:38.000000000 +0000
@@ -1451,6 +1451,12 @@
       removelist="$lobj $libobj ${libobj}T"
     fi
 
+    case $host_os in
+    kiwi*)
+      pic_mode=yes
+      ;;
+    esac
+
     # On Cygwin there's no "real" PIC flag so we must build both object types
     case $host_os in
     cygwin* | mingw* | pw32* | os2* | cegcc*)
