diff -urN gcc-4.8.1.orig/config.sub gcc-4.8.1/config.sub
--- gcc-4.8.1.orig/config.sub	2013-01-11 13:02:58.000000000 +0000
+++ gcc-4.8.1/config.sub	2013-07-04 16:47:37.000000000 +0100
@@ -1378,7 +1378,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -kiwi*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN gcc-4.8.1.orig/gcc/config/arm/kiwi.h gcc-4.8.1/gcc/config/arm/kiwi.h
--- gcc-4.8.1.orig/gcc/config/arm/kiwi.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.8.1/gcc/config/arm/kiwi.h	2013-07-04 16:47:37.000000000 +0100
@@ -0,0 +1,43 @@
+/* Definitions for Kiwi ARM targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+/* We want both the BPABI builtins and the Kiwi builtins. */
+#undef  TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS() 		\
+  do 						\
+    {						\
+      TARGET_BPABI_CPP_BUILTINS();		\
+      KIWI_TARGET_OS_CPP_BUILTINS();		\
+    }						\
+  while (false)
+
+#define KIWI_LINK_SPEC " \
+  %{shared:-shared} \
+  %{symbolic:-Bsymbolic} \
+  %{mbig-endian:-EB} %{mlittle-endian:-EL} \
+  -X \
+  %{!shared: \
+    %{!static: \
+      %{rdynamic:-export-dynamic} \
+      %{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}} \
+    %{static:-Bstatic}}"
+
+/* Include BPABI link spec as well as our own. */
+#undef  LINK_SPEC
+#define LINK_SPEC KIWI_LINK_SPEC BE8_LINK_SPEC
diff -urN gcc-4.8.1.orig/gcc/config/arm/t-kiwi gcc-4.8.1/gcc/config/arm/t-kiwi
--- gcc-4.8.1.orig/gcc/config/arm/t-kiwi	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.8.1/gcc/config/arm/t-kiwi	2013-07-04 16:47:37.000000000 +0100
@@ -0,0 +1,2 @@
+# These functions are included in shared libraries.
+TARGET_LIBGCC2_CFLAGS = -fPIC
diff -urN gcc-4.8.1.orig/gcc/config/i386/kiwi32.h gcc-4.8.1/gcc/config/i386/kiwi32.h
--- gcc-4.8.1.orig/gcc/config/i386/kiwi32.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.8.1/gcc/config/i386/kiwi32.h	2013-07-04 16:47:37.000000000 +0100
@@ -0,0 +1,28 @@
+/* Definitions for Kiwi i386 targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+#undef  LINK_SPEC
+#define LINK_SPEC "%{m64:-m elf_x86_64} %{!m64:-m elf_i386} \
+  %{shared:-shared} \
+  %{!shared: \
+    %{!static: \
+      %{rdynamic:-export-dynamic} \
+      %{m64:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}} \
+      %{!m64:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}}} \
+    %{static:-static}}"
diff -urN gcc-4.8.1.orig/gcc/config/i386/kiwi64.h gcc-4.8.1/gcc/config/i386/kiwi64.h
--- gcc-4.8.1.orig/gcc/config/i386/kiwi64.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.8.1/gcc/config/i386/kiwi64.h	2013-07-04 16:47:37.000000000 +0100
@@ -0,0 +1,28 @@
+/* Definitions for Kiwi x86-64 targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+#undef  LINK_SPEC
+#define LINK_SPEC "%{!m32:-m elf_x86_64} %{m32:-m elf_i386} \
+  %{shared:-shared} \
+  %{!shared: \
+    %{!static: \
+      %{rdynamic:-export-dynamic} \
+      %{!m32:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}} \
+      %{m32:%{!dynamic-linker:-dynamic-linker /system/libraries/libkernel.so}}} \
+    %{static:-static}}"
diff -urN gcc-4.8.1.orig/gcc/config/i386/t-kiwi64 gcc-4.8.1/gcc/config/i386/t-kiwi64
--- gcc-4.8.1.orig/gcc/config/i386/t-kiwi64	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.8.1/gcc/config/i386/t-kiwi64	2013-07-04 16:47:37.000000000 +0100
@@ -0,0 +1,3 @@
+MULTILIB_OPTIONS = m64/m32
+MULTILIB_DIRNAMES = 64 32
+MULTILIB_OSDIRNAMES = ../lib ../lib32
diff -urN gcc-4.8.1.orig/gcc/config/kiwi.h gcc-4.8.1/gcc/config/kiwi.h
--- gcc-4.8.1.orig/gcc/config/kiwi.h	1970-01-01 01:00:00.000000000 +0100
+++ gcc-4.8.1/gcc/config/kiwi.h	2013-07-04 16:47:37.000000000 +0100
@@ -0,0 +1,44 @@
+/* Definitions for Kiwi targets
+   Copyright (C) 2007, Free Software Foundation, Inc.
+
+This file is part of GCC.
+
+GCC is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3, or (at your option)
+any later version.
+
+GCC is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GCC; see the file COPYING3.  If not see
+<http://www.gnu.org/licenses/>.  */
+
+/* Specify predefined symbols in preprocessor.  */
+#undef  KIWI_TARGET_OS_CPP_BUILTINS
+#define KIWI_TARGET_OS_CPP_BUILTINS()		\
+	do {					\
+		builtin_define ("__Kiwi__");	\
+	} while (0)
+
+#undef  TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()		\
+	KIWI_TARGET_OS_CPP_BUILTINS()
+
+#undef  STARTFILE_SPEC
+#define STARTFILE_SPEC \
+  "%{!shared: %{pg|p|profile:gcrt1.o%s;pie:Scrt1.o%s;:crt1.o%s}} \
+   crti.o%s %{static:crtbeginT.o%s;shared|pie:crtbeginS.o%s;:crtbegin.o%s}"
+
+#undef  ENDFILE_SPEC
+#define ENDFILE_SPEC \
+  "%{shared|pie:crtendS.o%s;:crtend.o%s} crtn.o%s"
+
+#undef  LIB_SPEC
+#define LIB_SPEC "-lc"
+
+/* Make no assumptions about header files. */
+#define NO_IMPLICIT_EXTERN_C
diff -urN gcc-4.8.1.orig/gcc/config.gcc gcc-4.8.1/gcc/config.gcc
--- gcc-4.8.1.orig/gcc/config.gcc	2013-03-13 09:47:41.000000000 +0000
+++ gcc-4.8.1/gcc/config.gcc	2013-07-04 16:47:37.000000000 +0100
@@ -810,6 +810,12 @@
   default_use_cxa_atexit=yes
   use_gcc_stdint=wrap
   ;;
+*-*-kiwi*)
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  tmake_file="t-svr4"
+  ;;
 esac
 
 case ${target} in
@@ -902,7 +908,7 @@
 	# The EABI requires the use of __cxa_atexit.
 	default_use_cxa_atexit=yes
 	;;
-arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems*)
+arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems* | arm*-*-kiwi*)
 	case ${target} in
 	arm*eb-*-eabi*)
 	  tm_defines="${tm_defines} TARGET_BIG_ENDIAN_DEFAULT=1"
@@ -920,6 +926,10 @@
 	  tmake_file="${tmake_file} arm/t-bpabi"
 	  use_gcc_stdint=wrap
 	  ;;
+	arm*-*-kiwi*)
+	  tm_file="${tm_file} kiwi.h arm/kiwi.h"
+	  tmake_file="${tmake_file} arm/t-bpabi"
+	  ;;
 	arm*-*-rtems*)
 	  tm_file="${tm_file} rtems.h arm/rtems-eabi.h newlib-stdint.h"
 	  tmake_file="${tmake_file} arm/t-bpabi t-rtems arm/t-rtems-eabi"
@@ -1549,6 +1559,13 @@
 		tm_file="${tm_file} dbxcoff.h"
 	fi
 	;;
+i[34567]86-*-kiwi*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h kiwi.h i386/kiwi32.h"
+	;;
+x86_64-*-kiwi*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h kiwi.h i386/x86-64.h i386/kiwi64.h"
+	tmake_file="${tmake_file} i386/t-kiwi64"
+	;;
 ia64*-*-elf*)
 	tm_file="${tm_file} dbxelf.h elfos.h newlib-stdint.h ia64/sysv4.h ia64/elf.h"
 	tmake_file="ia64/t-ia64"
diff -urN gcc-4.8.1.orig/libgcc/config.host gcc-4.8.1/libgcc/config.host
--- gcc-4.8.1.orig/libgcc/config.host	2013-02-04 19:06:20.000000000 +0000
+++ gcc-4.8.1/libgcc/config.host	2013-07-04 16:47:37.000000000 +0100
@@ -195,6 +195,10 @@
       ;;
   esac
   ;;
+*-*-kiwi*)
+  tmake_file="$tmake_file t-crtstuff-pic t-libgcc-pic"
+  extra_parts="crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o"
+  ;;
 *-*-linux* | frv-*-*linux* | *-*-kfreebsd*-gnu | *-*-knetbsd*-gnu | *-*-gnu* | *-*-kopensolaris*-gnu)
   tmake_file="$tmake_file t-crtstuff-pic t-libgcc-pic t-eh-dw2-dip t-slibgcc t-slibgcc-gld t-slibgcc-elf-ver t-linux"
   extra_parts="crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o"
@@ -337,7 +341,7 @@
 	tmake_file="$tmake_file arm/t-arm arm/t-elf t-softfp-sfdf t-softfp-excl arm/t-softfp t-softfp"
 	extra_parts="$extra_parts crti.o crtn.o"
 	;;
-arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems*)
+arm*-*-eabi* | arm*-*-symbianelf* | arm*-*-rtems* | arm*-*-kiwi*)
 	tmake_file="${tmake_file} arm/t-arm arm/t-elf t-fixedpoint-gnu-prefix"
 	tm_file="$tm_file arm/bpabi-lib.h"
 	case ${host} in
@@ -345,6 +349,9 @@
 	  tmake_file="${tmake_file} arm/t-bpabi"
 	  extra_parts="crtbegin.o crtend.o crti.o crtn.o"
 	  ;;
+	arm*-*-kiwi*)
+	  tmake_file="${tmake_file} arm/t-bpabi"
+	  ;;
 	arm*-*-symbianelf*)
 	  tmake_file="${tmake_file} arm/t-symbian t-slibgcc-nolc-override"
 	  tm_file="$tm_file arm/symbian-lib.h"
@@ -507,6 +514,12 @@
 x86_64-*-freebsd*)
 	tmake_file="${tmake_file} i386/t-freebsd i386/t-crtstuff"
 	;;
+i[34567]86-*-kiwi*)
+	tmake_file="${tmake_file} i386/t-crtstuff"
+	;;
+x86_64-*-kiwi*)
+	tmake_file="${tmake_file} i386/t-crtstuff"
+	;;
 i[34567]86-*-netbsdelf*)
 	;;
 x86_64-*-netbsd*)
diff -urN gcc-4.8.1.orig/libstdc++-v3/configure.ac gcc-4.8.1/libstdc++-v3/configure.ac
--- gcc-4.8.1.orig/libstdc++-v3/configure.ac	2013-04-09 21:47:55.000000000 +0100
+++ gcc-4.8.1/libstdc++-v3/configure.ac	2013-07-04 16:53:26.000000000 +0100
@@ -115,9 +115,18 @@
   glibcxx_compiler_shared_flag="-D_GLIBCXX_SHARED"
 
 else
-  glibcxx_lt_pic_flag=
-  glibcxx_compiler_pic_flag=
-  glibcxx_compiler_shared_flag=
+  case $host_os in
+  kiwi*)
+    glibcxx_lt_pic_flag="-prefer-pic"
+    glibcxx_compiler_pic_flag="$lt_prog_compiler_pic_CXX"
+    glibcxx_compiler_shared_flag="-D_GLIBCXX_SHARED"
+    ;;
+  *)
+    glibcxx_lt_pic_flag=
+    glibcxx_compiler_pic_flag=
+    glibcxx_compiler_shared_flag=
+    ;;
+  esac
 fi
 AC_SUBST(glibcxx_lt_pic_flag)
 AC_SUBST(glibcxx_compiler_pic_flag)
diff -urN gcc-4.8.1.orig/libstdc++-v3/configure.host gcc-4.8.1/libstdc++-v3/configure.host
--- gcc-4.8.1.orig/libstdc++-v3/configure.host	2012-10-15 22:12:23.000000000 +0100
+++ gcc-4.8.1/libstdc++-v3/configure.host	2013-07-04 16:47:37.000000000 +0100
@@ -275,6 +275,10 @@
   hpux*)
     os_include_dir="os/hpux"
     ;;
+  kiwi*)
+    error_constants_dir="os/generic"
+    os_include_dir="os/generic"
+    ;;
   mingw32*)
     case "$host" in
       *-w64-*)
diff -urN gcc-4.8.1.orig/libstdc++-v3/crossconfig.m4 gcc-4.8.1/libstdc++-v3/crossconfig.m4
--- gcc-4.8.1.orig/libstdc++-v3/crossconfig.m4	2012-05-02 20:34:31.000000000 +0100
+++ gcc-4.8.1/libstdc++-v3/crossconfig.m4	2013-07-04 16:47:37.000000000 +0100
@@ -141,6 +141,10 @@
 	;;
     esac
     ;;
+  *-kiwi*)
+    GLIBCXX_CHECK_COMPILER_FEATURES
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    ;;
   *-linux* | *-uclinux* | *-gnu* | *-kfreebsd*-gnu | *-knetbsd*-gnu)
     GLIBCXX_CHECK_COMPILER_FEATURES
     GLIBCXX_CHECK_LINKER_FEATURES
diff -urN gcc-4.8.1.orig/libtool.m4 gcc-4.8.1/libtool.m4
--- gcc-4.8.1.orig/libtool.m4	2012-09-15 00:23:41.000000000 +0100
+++ gcc-4.8.1/libtool.m4	2013-07-04 16:47:37.000000000 +0100
@@ -5371,7 +5371,9 @@
   _LT_LINKER_SHLIBS($1)
   _LT_SYS_DYNAMIC_LINKER($1)
   _LT_LINKER_HARDCODE_LIBPATH($1)
-  LT_SYS_DLOPEN_SELF
+  if test "$cross_compiling" = no; then
+    LT_SYS_DLOPEN_SELF
+  fi
   _LT_CMD_STRIPLIB
 
   # Report which library types will actually be built
