diff -urN gcc-4.6.1.orig/gcc/Makefile.in gcc-4.6.1/gcc/Makefile.in
--- gcc-4.6.1.orig/gcc/Makefile.in	2011-07-05 14:31:18.808153773 +0100
+++ gcc-4.6.1/gcc/Makefile.in	2011-07-05 14:32:29.065079648 +0100
@@ -4168,7 +4168,7 @@
 	      gcc_dir=`${PWD_COMMAND}` ; \
 	      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
 	      cd $(build_objdir)/fixincludes && \
-	      $(SHELL) ./fixinc.sh "$${gcc_dir}/$${fix_dir}" \
+	      $(SHELL) -c true "$${gcc_dir}/$${fix_dir}" \
 	        $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
 	    rm -f $${fix_dir}/syslimits.h; \
 	    if [ -f $${fix_dir}/limits.h ]; then \
