diff -urN binutils-2.19.1.orig/bfd/config.bfd binutils-2.19.1/bfd/config.bfd
--- binutils-2.19.1.orig/bfd/config.bfd	2008-04-16 19:02:01.000000000 +0100
+++ binutils-2.19.1/bfd/config.bfd	2009-05-05 21:46:35.957294023 +0100
@@ -620,6 +620,11 @@
     want64=true
     targ_underscore=yes
     ;;
+  x86_64-*-kiwi*)
+    targ_defvec=bfd_elf64_x86_64_vec
+    targ_selvecs=bfd_elf32_i386_vec
+    want64=true
+    ;;
 #endif
   i[3-7]86-*-lynxos*)
     targ_defvec=bfd_elf32_i386_vec
@@ -681,6 +686,9 @@
     targ_defvec=bfd_elf32_i386_vec
     targ_selfvecs=i386chaos_vec
     ;;
+  i[3-7]86-*-kiwi*)
+    targ_defvec=bfd_elf32_i386_vec
+    ;;
 
   i860-*-mach3* | i860-*-osf1* | i860-*-coff*)
     targ_defvec=i860coff_vec
diff -urN binutils-2.19.1.orig/config.sub binutils-2.19.1/config.sub
--- binutils-2.19.1.orig/config.sub	2008-04-14 10:28:35.000000000 +0100
+++ binutils-2.19.1/config.sub	2009-05-05 17:11:59.692713282 +0100
@@ -1266,7 +1266,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -kiwi*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN binutils-2.19.1.orig/gas/configure.tgt binutils-2.19.1/gas/configure.tgt
--- binutils-2.19.1.orig/gas/configure.tgt	2008-10-02 09:07:17.000000000 +0100
+++ binutils-2.19.1/gas/configure.tgt	2009-05-05 18:03:45.596798102 +0100
@@ -178,6 +178,7 @@
   i386-*-coff)				fmt=coff ;;
   i386-*-elf)				fmt=elf ;;
   i386-*-kaos*)				fmt=elf ;;
+  i386-*-kiwi*)				fmt=elf ;;
   i386-*-bsd*)				fmt=aout em=386bsd ;;
   i386-*-netbsd0.8)			fmt=aout em=386bsd ;;
   i386-*-netbsdpe*)			fmt=coff em=pe ;;
diff -urN binutils-2.19.1.orig/ld/configure.tgt binutils-2.19.1/ld/configure.tgt
--- binutils-2.19.1.orig/ld/configure.tgt	2008-08-09 06:35:13.000000000 +0100
+++ binutils-2.19.1/ld/configure.tgt	2009-05-05 20:30:33.353294711 +0100
@@ -240,6 +240,10 @@
 			targ_extra_emuls=elf_i386
 			;;
 i[3-7]86-*-kaos*)	targ_emul=elf_i386 ;;
+i[3-7]86-*-kiwi*)	targ_emul=elf_i386 ;;
+x86_64-*-kiwi*)	targ_emul=elf_x86_64
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-freebsdaout* | i[3-7]86-*-freebsd[12].* | i[3-7]86-*-freebsd[12])
 			targ_emul=i386bsd ;;
 i[3-7]86-*-freebsd* | i[3-7]86-*-kfreebsd*-gnu | i[3-7]86-*-dragonfly*)
