diff -urN gcc-4.4.2.orig/gcc/Makefile.in gcc-4.4.2/gcc/Makefile.in
--- gcc-4.4.2.orig/gcc/Makefile.in	2009-10-15 18:07:07.458827060 +0100
+++ gcc-4.4.2/gcc/Makefile.in	2009-10-15 18:07:51.315052029 +0100
@@ -3701,7 +3701,7 @@
 	      SHELL='$(SHELL)'; MACRO_LIST=`${PWD_COMMAND}`/macro_list ; \
 	      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
 	      cd $(build_objdir)/fixincludes && \
-	      $(SHELL) ./fixinc.sh ../../gcc/$${fix_dir} \
+	      $(SHELL) -c true ../../gcc/$${fix_dir} \
 	        $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
 	    rm -f $${fix_dir}/syslimits.h; \
 	    if [ -f $${fix_dir}/limits.h ]; then \
