diff -urN gcc-4.5.2.orig/gcc/Makefile.in gcc-4.5.2/gcc/Makefile.in
--- gcc-4.5.2.orig/gcc/Makefile.in	2010-12-22 09:46:00.422295546 +0000
+++ gcc-4.5.2/gcc/Makefile.in	2010-12-22 09:46:59.815028919 +0000
@@ -3953,7 +3953,7 @@
 	      gcc_dir=`${PWD_COMMAND}` ; \
 	      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
 	      cd $(build_objdir)/fixincludes && \
-	      $(SHELL) ./fixinc.sh "$${gcc_dir}/$${fix_dir}" \
+	      $(SHELL) -c true "$${gcc_dir}/$${fix_dir}" \
 	        $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
 	    rm -f $${fix_dir}/syslimits.h; \
 	    if [ -f $${fix_dir}/limits.h ]; then \
